### PR TITLE
Validate if push constants doesn't update and image subresources are used in attachments and descriptor sets

### PR DESCRIPTION
**Pull request by locke-lunarg**
```
VUID-vkCmdDraw-None-02698
VUID-vkCmdDrawIndexed-None-02698
VUID-vkCmdDrawIndirect-None-02698
VUID-vkCmdDrawIndirectCount-None-02698
VUID-vkCmdDrawIndexedIndirect-None-02698
VUID-vkCmdDrawIndexedIndirectCount-None-02698
VUID-vkCmdDrawIndirectByteCountEXT-None-02698
VUID-vkCmdDrawMeshTasksNV-None-02698
VUID-vkCmdDrawMeshTasksIndirectNV-None-02698
VUID-vkCmdDrawMeshTasksIndirectCountNV-None-02698
VUID-vkCmdDispatch-None-02698
VUID-vkCmdDispatchIndirect-None-02698
VUID-vkCmdDispatchBase-None-02698
VUID-vkCmdExecuteGeneratedCommandsNV-None-02698
VUID-vkCmdTraceRaysNV-None-02698
VUID-vkCmdTraceRaysKHR-None-02698
VUID-vkCmdTraceRaysIndirectKHR-None-02698

For each push constant that is statically used by the VkPipeline bound to the pipeline bind point used by this command, a push constant value must have been set for the same pipeline bind point, with a VkPipelineLayout that is compatible for push constants, with the VkPipelineLayout used to create the current VkPipeline, as described in Pipeline Layout Compatibility
```
#135 & #306 
Check if update of push constants of VkPipelinLayout is compatible with binding VkPipelineLayout of VkPipeline. Plus, check if every byte of push constants update.

### CODE REVIEW

**Comment by locke-lunarg** [comment]
```
VUID-vkCmdDraw-None-02687
VUID-vkCmdDrawIndexed-None-02687
VUID-vkCmdDrawIndirect-None-02687
VUID-vkCmdDrawIndirectCount-None-02687
VUID-vkCmdDrawIndexedIndirect-None-02687
VUID-vkCmdDrawIndexedIndirectCount-None-02687
VUID-vkCmdDrawIndirectByteCountEXT-None-02687
VUID-vkCmdDrawMeshTasksNV-None-02687
VUID-vkCmdDrawMeshTasksIndirectNV-None-02687
VUID-vkCmdDrawMeshTasksIndirectCountNV-None-02687
Image subresources used as attachments in the current render pass must not be accessed in any way other than as an attachment by this command
```
#135
Check if a image view or image subresoucse is used in both attachments in current renderpass and current subpass and descriptor sets of binding.  It doesn't only check in vkCmdDraw, also vkQueueSubmit becuase of VK_DESCRIPTOR_BINDING_PARTIALLY_BOUND_BIT_EXT.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1174,6 +1176,24 @@ bool CoreChecks::ValidateCmdBufDrawState(const CMD_BUFFER_STATE *cb_node, CMD_TY
     if (VK_PIPELINE_BIND_POINT_GRAPHICS == bind_point)
         result |= ValidatePipelineDrawtimeState(state, cb_node, cmd_type, pPipe, function);
 
+    // Verify if push constants have been set
+    if (pipeline_layout->push_constant_ranges != cb_node->push_constant_data_ranges) {
+        result |= LogError(cb_node->commandBuffer, vuid.push_constants_set,
```
Comment (mark-lunarg) [inline layers/core_validation.cpp:1181]: Any message that outputs more than a single handle needs to use a `LogObjectList()`.  See other examples in this file.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -7962,12 +7984,14 @@ bool CoreChecks::PreCallValidateCmdPushConstants(VkCommandBuffer commandBuffer,
...
-                                     "),  must contain all stages in overlapping VkPushConstantRange stageFlags (0x%" PRIx32
-                                     "), offset (%" PRIu32 "), and size (%" PRIu32 ") in %s.",
-                                     (uint32_t)stageFlags, offset, size, (uint32_t)range.stageFlags, range.offset, range.size,
-                                     report_data->FormatHandle(layout).c_str());
+                    skip |=
+                        LogError(commandBuffer, "VUID-vkCmdPushConstants-offset-01796",
+                                 "vkCmdPushConstants(): stageFlags (%s, offset (%" PRIu32 "), and size (%" PRIu32
+                                 "),  must contain all stages in overlapping VkPushConstantRange stageFlags (%s), offset (%" PRIu32
+                                 "), and size (%" PRIu32 ") in %s.",
+                                 string_VkShaderStageFlags(stageFlags).c_str(), offset, size,
```
Comment (mark-lunarg) [inline layers/core_validation.cpp:7992]: Nice!  Much better...

File: layers/core_validation.cpp (hunk 1 above)
Comment (locke-lunarg) [reply layers/core_validation.cpp]: fixed it

**Pull request merged**